lib/crypto/crypto_compat.c

```diff
@@ -26,3 +26,37 @@ crypto_compat_RSA_valid_size(RSA * rsa)
 	/* Failure! */
 	return (-1);
 }
+
+/**
+ * crypto_compat_RSA_import(key, n, e, d, p, q, dmp1, dmq1, iqmp):
+ * Import the given BIGNUMs into the RSA ${key}.
+ */
+int
+crypto_compat_RSA_import(RSA ** key, BIGNUM * n, BIGNUM * e, BIGNUM * d,
+    BIGNUM * p, BIGNUM * q, BIGNUM * dmp1, BIGNUM * dmq1, BIGNUM * iqmp)
+{
+
+	/* Sanity checks. */
+	assert(key != NULL);
+	assert((n != NULL) && (e != NULL));
+	/* All the private-key-related variables are NULL, or they're not. */
+	assert( ((d == NULL) && (p == NULL) && (q == NULL) && (dmp1 == NULL)
```

Comment:
Not certain what formatting you'd prefer here.  I personally would split up the nesting even more to make it more obvious what's happening, but I suspect you might tell me to remove all semantic spacing entirely.